packages/sdk/src/Configuration/Policies/NoDepegOnRedeemSharesForSpecificAssets.ts

```diff
@@ -0,0 +1,70 @@
+import { type Address, type Hex, decodeAbiParameters, encodeAbiParameters } from "viem";
+import { Assertion } from "../../Utils.js";
+
+const settingsEncoding = [
+  {
+    type: "address[]",
+    name: "assets",
+  },
+  {
+    type: "address[]",
+    name: "referenceAssets",
+  },
+  {
+    type: "uint16[]",
+    name: "deviationToleranceInBpss",
+  },
+] as const;
+
+export type Settings = {
+  assetConfigs: ReadonlyArray<{
+    /**
+     * The asset that is pegged to the reference asset.
+     */
+    asset: Address;
+    /**
+     * The reference asset that the asset is pegged to.
+     */
+    referenceAsset: Address;
+    /**
+     * The deviation tolerance in basis points between the pegged asset and its reference asset.
+     */
+    deviationToleranceInBps: number;
+  }>;
+};
+
+/**
+ * Encodes the given settings into a hex string.
+ *
+ * @returns The encoded settings.
+ */
+export function encodeSettings(args: Settings): Hex {
+  const assets = args.assetConfigs.map(({ asset }) => asset);
+  const referenceAssets = args.assetConfigs.map(({ referenceAsset }) => referenceAsset);
+  const deviationToleranceInBpss = args.assetConfigs.map(({ deviationToleranceInBps }) => deviationToleranceInBps);
```

Comment:
It won't make much of a difference, but instead of iterating through the `args.assetConfigs` array three time, you could do one for loop.